diff -Nru gdb-9.1/gdb/remote.c gdb_patched/gdb/remote.c
--- gdb-9.1/gdb/remote.c	2020-02-08 23:50:14.000000000 +1100
+++ gdb_patched/gdb/remote.c	2020-03-18 22:07:20.371358253 +1100
@@ -7979,24 +7979,17 @@
 
   buf_len = strlen (rs->buf.data ());
 
-  /* Further sanity checks, with knowledge of the architecture.  */
-  if (buf_len > 2 * rsa->sizeof_g_packet)
-    error (_("Remote 'g' packet reply is too long (expected %ld bytes, got %d "
-	     "bytes): %s"),
-	   rsa->sizeof_g_packet, buf_len / 2,
-	   rs->buf.data ());
-
   /* Save the size of the packet sent to us by the target.  It is used
      as a heuristic when determining the max size of packets that the
      target can safely receive.  */
-  if (rsa->actual_register_packet_size == 0)
+  if (rsa->actual_register_packet_size != buf_len)
     rsa->actual_register_packet_size = buf_len;
 
   /* If this is smaller than we guessed the 'g' packet would be,
      update our records.  A 'g' reply that doesn't include a register's
      value implies either that the register is not available, or that
      the 'p' packet must be used.  */
-  if (buf_len < 2 * rsa->sizeof_g_packet)
+  if (buf_len != 2 * rsa->sizeof_g_packet)
     {
       long sizeof_g_packet = buf_len / 2;
 
